sklearn/linear_model/_ridge.py

```diff
@@ -795,22 +806,72 @@ def _ridge_regression(
         )
 
     if solver == "svd":
-        if sparse.issparse(X):
+        if X_is_sparse:
             raise TypeError("SVD solver does not support sparse inputs currently")
-        coef = _solve_svd(X, y, alpha)
+        coef = _solve_svd(X, y, alpha, xp)
 
     if ravel:
-        # When y was passed as a 1d-array, we flatten the coefficients.
-        coef = coef.ravel()
+        coef = _ravel(coef)
+
+    coef = xp.asarray(coef)
 
     if return_n_iter and return_intercept:
-        return coef, n_iter, intercept
+        res = coef, n_iter, intercept
     elif return_intercept:
-        return coef, intercept
+        res = coef, intercept
     elif return_n_iter:
-        return coef, n_iter
+        res = coef, n_iter
     else:
-        return coef
+        res = coef
+
+    return (*res, solver) if return_solver else res
+
+
+def _solver_auto_set(solver, positive, return_intercept, is_sparse, xp):
```

Comment:
I'm not a great fan of the function name. "set" makes me think it does something with `set()`, because it is at the end of the name, but then I think maybe it actually sets something (like a setter does).

I think what the function actually does is resolve the `"auto"` value to a concrete name of a solver. Maybe `resolve_solver`, `resolve_solver_option` or `check_solver_choice` or something like this would make it easier to look at the name and know what it does. (For extra bonus points we could add a docstring, but I think even with a docstring having a good name is important)